core/src/main/java/com/linecorp/armeria/server/AbstractHttpResponseHandler.java

```diff
@@ -111,7 +112,10 @@ final boolean failIfStreamOrSessionClosed() {
         // 3. After successfully flushing, the listener requests next data and
         //    the subscriber attempts to write the next data to the stream closed at 2).
         if (!isWritable()) {
-            Throwable cause = CapturedServiceException.get(reqCtx);
+            Throwable cause = null;
+            if (reqCtx.log().isAvailable(RequestLogProperty.RESPONSE_CAUSE)) {
+                cause = reqCtx.log().ensureAvailable(RequestLogProperty.RESPONSE_CAUSE).responseCause();
```

Comment:
Side note for Kotlin: It looks verbose to use `isAvailable(RequestLogProperty.RESPONSE_CAUSE)` and `ensureAvailable(RequestLogProperty.RESPONSE_CAUSE)` to get the value when it is available.

The following style might be useful to take advantage of Kotlin null [safe calls](https://kotlinlang.org/docs/null-safety.html#safe-calls).
```kt
val cause: Throwable? = reqCtx.log().getIfAvailable(RequestLogProperty.RESPONSE_CAUSE)?.responseCause()
```

Reply:
Care to create an issue labled with `good-first-issue`?

Reply:
Sure!